src/parser/ml/FStar_Parser_LexFStar.ml

```diff
@@ -304,141 +304,146 @@ let push_one_line_comment pre lexbuf =
+let u_mod = [%sedlex.regexp? mc]
 (** Me *)
-let regexp u_mod_enclose = [0x0488-0x0489 0x1abe 0x20dd-0x20e0 0x20e2-0x20e4 0xa670-0xa672]
+let u_mod_enclose = [%sedlex.regexp? me]
 (** Cc *)
-let regexp u_ascii_control = [0x0000-0x001f 0x007f-0x009f]
+let u_ascii_control = [%sedlex.regexp? cc]
 (** Cf *)
-let regexp u_format_control = [0x00ad 0x0600-0x0605 0x061c 0x06dd 0x070f 0x180e 0x200b-0x200f 0x202a-0x202e 0x2060-0x2064 0x2066-0x206f 0xfeff 0xfff9-0xfffb 0x110bd 0x1bca0-0x1bca3 0x1d173-0x1d17a 0xe0001 0xe0020-0xe007f]
+let u_format_control = [%sedlex.regexp? cf]
 (** Co *)
-let regexp u_private_use = [0xe000 0xf8ff 0xf0000 0xffffd 0x100000 0x10fffd]
+let u_private_use = [%sedlex.regexp? co]
 (** Cs *)
-let regexp u_surrogate = [0xd800 0xdb7f-0xdb80 0xdbff-0xdc00 0xdfff]
+let u_surrogate = [%sedlex.regexp? cs]
 
 (* -------------------------------------------------------------------- *)
-let regexp lower  = u_lower
-let regexp upper  = u_upper | u_title
-let regexp letter = u_lower | u_upper | u_other | u_modifier
-let regexp digit  = ['0'-'9']
-let regexp hex    = ['0'-'9'] | ['A'-'F'] | ['a'-'f']
+let lower  = [%sedlex.regexp? u_lower]
+let upper  = [%sedlex.regexp? u_upper | u_title]
+let letter = [%sedlex.regexp? u_lower | u_upper | u_other | u_modifier]
+let digit  = [%sedlex.regexp? '0'..'9']
+let hex    = [%sedlex.regexp? '0'..'9' | 'A'..'F' | 'a'..'f']
 
 (* -------------------------------------------------------------------- *)
-let regexp anywhite  = u_space | u_space_extra
-let regexp newline   = "\r\n" | [10 13 0x2028 0x2029]
+let anywhite  = [%sedlex.regexp? u_space | u_space_extra]
+let newline   = [%sedlex.regexp? "\r\n" | 10 | 13 | 0x2028 | 0x2029]
 
 (* -------------------------------------------------------------------- *)
-let regexp op_char = ["!$%&*+-./<=?^|~:"]
-let regexp ignored_op_char = [".$"]
-let regexp op_token =
-  "~" | "-" | "/\\" | "\\/" | "<:" | "<@" | "(|" | "|)" | "#" |
-  "u#" | "&" | "()" | "(" | ")" | "," | "~>" | "->" | "<--" |
-  "<-" | "<==>" | "==>" | "." | "?." | "?" | ".[|" | ".[" | ".(|" | ".(" |
-  "$" | "{:pattern" | ":" | "::" | ":=" | ";;" | ";" | "=" | "%[" |
-  "!{" | "[@@" | "[@" | "[|" | "{|" | "[" | "|>" | "]" | "|]" | "|}" | "{" | "|" | "}"
+let op_char = [%sedlex.regexp? Chars "!$%&*+-./<=?^|~:"]
+let ignored_op_char = [%sedlex.regexp? Chars ".$"]
+
+(* op_token must be splt into seperate regular expressions to prevent
+   compliation from hanging *)
```

Comment:
that's interesting ... can you comment about this?

Reply:
How did you diagnose that this particular regexp was the source of the hanging problem?

Reply:
I indeed see it hanging on

00:00:58 88   (87  ) ...parser/ml/FStar_Parser_LexFStar.ml.depends   O-b----- \
00:00:58 88   (87  ) ...parser/ml/FStar_Parser_LexFStar.ml.depends   O-b----- \
00:00:59 88   (87  ) ...parser/ml/FStar_Parser_LexFStar.ml.depends

Reply:
> How did you diagnose that this particular regexp was the source of the hanging problem?

It was mostly just trial and error changing things until the code compiled :)

I am not familiar with sedlex internals so I don't know the exact reason for this problem. When I have some free time, I will try to create a minimal repro and ask the sedlex maintainers.

Reply:
I opened an issue in the sedlex repository here: https://github.com/ocaml-community/sedlex/issues/97

Reply:
thank you! This is *very* helpful